uniffi/src/templates/ObjectTemplate.rs

```diff
@@ -6,41 +6,47 @@
 // If the caller's implementation of the struct does not match with the methods or types specified
 // in the IDL, then the rust compiler will complain with a (hopefully at least somewhat helpful!)
 // error message when processing this generated code.
-
+{% let handle_map = format!("UNIFFI_HANDLE_MAP_{}", obj.name().to_uppercase()) %}
 lazy_static::lazy_static! {
-    static ref UNIFFI_HANDLE_MAP_{{ obj.name()|upper }}: ffi_support::ConcurrentHandleMap<{{ obj.name() }}> = ffi_support::ConcurrentHandleMap::new();
+    static ref {{ handle_map }}: ffi_support::ConcurrentHandleMap<{{ obj.name() }}> = ffi_support::ConcurrentHandleMap::new();
 }
 
-// XXX TODO: destructors.
-// These will need to be defined as another FFI function on the Object struct, and included automatically
-// in the set of all FFI functions for use by the bindings.
-// define_handle_map_deleter!(UNIFFI_HANDLE_MAP_{{ obj.name() }}, {{ obj.name() }}_free);
+    // define_handle_map_deleter!(UNIFFI_HANDLE_MAP_{{ obj.name() }}, {{ obj.name() }}_free);
+    {% let ffi_free = obj.reserved_object_free().ffi_func() -%}
+    #[no_mangle]
+    pub extern "C" fn {{ ffi_free.name() }}(
+        {%- call rs::arg_list_rs_decl(ffi_free.arguments()) %}) {
+        log::debug!("{{ ffi_free.name() }}");
+        let _retval = {{ handle_map }}.delete_u64(handle);
+    }
+
```

Comment:
This should be replaced with 

```rust
define_handle_map_deleter!({{ handle_map }}, {{ ffi_free.name() }})
```

once #31 lands and I can cargo cult the how that works better.